src/components/DoctorsPage/DoctorsCard/Schedule/Schedule.tsx

```diff
@@ -1,74 +1,58 @@
-import React, { useReducer, useState } from 'react';
+import React, { useReducer } from 'react';
+import { useDispatch } from 'react-redux';
+import { useTypesSelector } from '@hooks/UseTypedSelector';
+import { ScheduleActionTypes } from './types';
+import { profilePatientAddAppointment } from '@store/actionCreators/profileData';
+import { ScheduleWrapper, WrapperHeader, ItemWrapper, SuccessMessage } from './ScheduleStyle';
 import Button from '@components/common/Button/Button';
 import ScheduleItem from './ScheduleItem/ScheduleItem';
-import { ScheduleWrapper, WrapperHeader, ItemWrapper, SuccessMessage } from './ScheduleStyle';
-import { ActionType, IScheduleInitialState, ScheduleActionTypes } from './types';
+import Error from '@components/common/Error/Error';
+import { initialState, reducer } from './ScheduleReducer';
 
 export interface IScheduleProps {
-  workTimeData: { date: string; time: string }[];
+  workTimeData: { date: string; time: string; _id: string }[];
 }
 
-const initialState = {
-  choiceWorkTime: {
-    date: '',
-    time: '',
-  },
-  disabledItem: null,
-  zeroing: false,
-  viewSuccessMessage: false,
-};
-
-const reducer = (state: IScheduleInitialState, action: ActionType): IScheduleInitialState => {
-  switch (action.type) {
-    case ScheduleActionTypes.SET_CHOICE_WORK_TIME:
-      return { ...state, choiceWorkTime: { date: action.payload.date, time: action.payload.time } };
-    case ScheduleActionTypes.SET_DISABLED_ITEM:
-      return { ...state, disabledItem: action.payload.disabledItem };
-    case ScheduleActionTypes.SUCCESS_MESSAGE:
-      return { ...state, viewSuccessMessage: true };
-    default:
-      return state;
-  }
-};
-
 const Schedule: React.FC<IScheduleProps> = ({ workTimeData }) => {
-  const [stateSchedule, dispatch] = useReducer(reducer, initialState);
-
+  const [stateSchedule, dispatchLock] = useReducer(reducer, initialState);
+  const { error } = useTypesSelector((state) => state.profile);
+  const dispatch = useDispatch();
+  
   const sendWorkTime = () => {
-    dispatch({ type: ScheduleActionTypes.SUCCESS_MESSAGE });
+    dispatchLock({ type: ScheduleActionTypes.SUCCESS_MESSAGE });
```

Comment:
name is quite easy to confuse with the redux reducer, maybe we could rename it?

Reply:
scheduleDispatch, renamed